lesion_analysis/time_point_lesion_evolution.py

```diff
@@ -0,0 +1,417 @@
+
+To do:
+    * Remove some of the outputed files
+
+Pierre-Louis Benveniste
+"""
+
+import os
+import argparse
+from pathlib import Path
+import nibabel as nib
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.cluster import DBSCAN
+import networkx as nx
+from scipy.optimize import linear_sum_assignment
+
+def get_parser():
+    """
+    This function parses the arguments given to the script.
+
+    Args:
+        None
+
+    Returns:
+        parser: parser containing the arguments
+    """
+
+    parser = argparse.ArgumentParser(description='Analyse the results of the segmentation of the MS lesion on the spinal cord.')
+    parser.add_argument('-i-1', '--input-first_image', required=True,
+                        help='Path to the first image')
+    parser.add_argument('-i-2', '--input-second_image', required=True,
+                        help='Path to the second image')
+    parser.add_argument('-seg-1', '--segmentation-first_image', required=True,
+                        help='Path to the lesion segmentation of the first image')
+    parser.add_argument('-seg-2', '--segmentation-second_image', required=True,
+                        help='Path to the lesion segmentation of the second image')
+    parser.add_argument('-o', '--output_folder', required=True,
+                        help='Path to the output folder')
+    parser.add_argument('--show', action='store_true',
+                        help='Whether to show the results or not')
+    return parser
+
+
+def perform_registration(path_first_image, path_second_image, path_first_lesion_seg, path_second_lesion_seg, path_output_folder):
+    """
+    This function performs the registration of the images and the segmentations using spinal cord toolbox
+
+    Args:
+        path_first_image: path to the first image
+        path_second_image: path to the second image
+        path_first_lesion_seg: path to the lesion segmentation of the first image
+        path_second_lesion_seg: path to the lesion segmentation of the second image
+        path_output_folder: path to the output folder
+    
+    Returns:
+        path_warp: path to the warp file
+        path_warpinv: path to the inverse warp file
+        path_second_image_registered: path to the second image registered
+        path_second_lesion_seg_registered: path to the second lesion segmentation registered
+    """
+
+    #first we segment the spinal cord on the two images using sct_deepseg_sc
+    os.system('sct_deepseg_sc -i ' + path_first_image + ' -c t2 -o ' + path_output_folder + '/first_image_sc_seg.nii.gz')
```

Comment:
system call can be avoided once (if) this script becomes an SCT function. Until then, it is fine to keep it as system call, but clear reference to SCT's version should be made in the README, as a requirement